src/main.c

```diff
@@ -699,9 +701,21 @@ static const char *listener_setup_ssl_picotls(struct listener_config_t *listener
         key = SSL_get_privatekey(fakeconn);
         assert(key != NULL);
         cert = SSL_get_certificate(fakeconn);
+        /* obtain peer verify mode */
+        verify_mode = (SSL_get_verify_mode(fakeconn) & SSL_VERIFY_PEER)? 1 : 0;
         SSL_free(fakeconn);
     }
 
+    if (verify_mode == 1) {
+        pctx->ctx.require_client_authentication = 1;
+        /* log the flag */
+        fprintf(stderr, "client cert verification is Enabled.\n");
+        /* set verify callback */
+        X509_STORE *ca_store = SSL_CTX_get_cert_store(ssl_ctx);
+        ptls_openssl_init_verify_certificate(&vc, ca_store);
```

Comment:
I'm not sure if we can call `ptls_openssl_init_verify_certificate` for every listener with mTLS enabled, taking same object (`vc`) as the argument. To paraphrase, I believe that `vc` has to be a member of `st_fat_context_t`.

Reply:
Agree. Need a callback per listener if mTLS enabled.

Reply:
updated based on your comment. Please review.